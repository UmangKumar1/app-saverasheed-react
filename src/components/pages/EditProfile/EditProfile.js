import React, { Component, Fragment } from 'react'
import styled from 'react-emotion'
import { prop } from 'ramda'
import PropTypes from 'prop-types'

import { userPropTypes } from '../../../fixtures/users'
import theme from '../../../theme'
import { containsBy, mapObjAsArray } from '../../../util/ramda-extra'
import Button from '../../atoms/Button'
import MenuPage from '../../templates/MenuPage'
import {
  googleAuth,
  facebookAuth,
  authIdPropTypes,
  portisAuth,
} from '../../../fixtures/authProviders'

const { func } = PropTypes

const {
  palette: {
    primary: { main: primaryMain, contrastText: primaryContrastText },
  },
} = theme

const ProfilePageContent = styled('div')({})

const SectionHeading = styled('h3')({
  padding: '2px 5px',
  display: 'inline-block',
  borderBottom: `solid 2px ${primaryMain}`,
})

const authProviders = [googleAuth, facebookAuth, portisAuth]

const IconWrapper = styled('div')({
  // background: 'white',
  padding: 8,
  display: 'flex',
  alignItems: 'center',
  marginRight: 8,
})

const SignInButtonRoot = styled(Button)(
  {
    marginBottom: 10,
  },
  ({ linked }) =>
    linked
      ? {
          color: 'white',
          backgroundColor: 'gray',
          border: 'solid 1px white',
        }
      : {
          color: primaryContrastText,
          backgroundColor: primaryMain,
          border: `solid 1px ${primaryMain}`,
        },
)

const SignInButton = ({
  auth: { name, id, Icon },
  linked,
  linking,
  ...props
}) => (
  <SignInButtonRoot linked={linked} {...props}>
    <IconWrapper>
      <Icon size={18} />
    </IconWrapper>
    <span>
      {linking
        ? `${linked ? 'Unlinking...' : 'Linking...'}`
        : `${linked ? 'Unlink' : 'Link'} ${name}`}
    </span>
  </SignInButtonRoot>
)

class EditProfile extends Component {
  static propTypes = {
    user: userPropTypes,
    unlinkAccount: func.isRequired,
    linkAccount: func.isRequired,
    linkingProvider: authIdPropTypes,
  }

  render() {
    const {
      user,
      unlinkAccount,
      linkAccount,
      linkingProvider,
      ...props
    } = this.props
    const providerData = prop('providerData', user)

    return (
      <MenuPage title="Edit Profile" {...props}>
        {!user ? (
          <div>Loading...</div>
        ) : (
          <Fragment>
            <ProfilePageContent>
              <SectionHeading>Accounts</SectionHeading>
              <div
                style={{
                  display: 'flex',
                  flexDirection: 'column',
                }}
              >
                {mapObjAsArray(auth => {
                  const { id } = auth
                  const linked = containsBy(
                    d => d.providerId === id,
                    providerData,
                  )
                  const clickFn = linked ? unlinkAccount : linkAccount
                  return (
                    <SignInButton
                      key={id}
                      onClick={() => clickFn(id)}
                      linking={linkingProvider === id}
                      {...{ auth, linked }}
                    />
                  )
                }, authProviders)}
              </div>
            </ProfilePageContent>
          </Fragment>
        )}
      </MenuPage>
    )
  }
}

export default EditProfile
