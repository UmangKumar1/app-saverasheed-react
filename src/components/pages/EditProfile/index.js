export { default, default as EditProfile } from './EditProfile'
