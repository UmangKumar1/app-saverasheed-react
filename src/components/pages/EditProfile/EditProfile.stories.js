import React from 'react'
import { storiesOf } from '@storybook/react'
import StoryRouter from 'storybook-react-router'

import EditProfile from '.'
import { user1 } from '../../../fixtures/users'

storiesOf('EditProfile', module)
  .addDecorator(StoryRouter())
  .add('default', () => <EditProfile user={user1} />)
