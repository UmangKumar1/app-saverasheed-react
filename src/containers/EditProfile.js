import { connect } from 'react-redux'

import EditProfile from '../components/pages/EditProfile'
import {
  linkAccountRequest,
  unlinkAccountRequest,
} from '../modules/auth/actions'
import { getLinkingProvider, getUser } from '../modules/auth/selectors'
import { subscribe } from '../util/reselect'

const mapStateToProps = subscribe({
  getUser,
  getLinkingProvider,
})

const actions = {
  unlinkAccount: unlinkAccountRequest,
  linkAccount: linkAccountRequest,
}

export default connect(
  mapStateToProps,
  actions,
)(EditProfile)
