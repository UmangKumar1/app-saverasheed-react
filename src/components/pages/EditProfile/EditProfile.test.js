import React from 'react'
import { shallow } from 'enzyme'
import EditProfile from '.'

const wrap = (props = {}) => shallow(<EditProfile {...props} />)

describe('EditProfile', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({ linkAccount: jest.fn(), unlinkAccount: jest.fn() })
    expect(wrapper).toMatchSnapshot()
  })
})
