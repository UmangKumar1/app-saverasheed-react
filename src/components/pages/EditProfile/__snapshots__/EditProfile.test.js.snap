// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditProfile
    linkAccount={[MockFunction]}
    unlinkAccount={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div>
        Loading...
      </div>,
      "title": "Edit Profile",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div>
          Loading...
        </div>,
        "title": "Edit Profile",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
